ldi-core/version-materialiser/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/VersionMaterialiser.java

```diff
@@ -0,0 +1,102 @@
+package be.vlaanderen.informatievlaanderen.ldes.ldi;
+
+import be.vlaanderen.informatievlaanderen.ldes.ldi.exceptions.ObjectIdentifierException;
+import be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiTransformer;
+import org.apache.jena.rdf.model.*;
+import org.apache.jena.rdf.model.impl.PropertyImpl;
+import org.apache.jena.rdf.model.impl.StatementImpl;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Map;
+
+public class VersionMaterialiser implements LdiTransformer {
+	private final Property versionPredicate;
+	private final boolean restrictToMembers;
+
+	public VersionMaterialiser(Property versionPredicate, boolean restrictToMembers) {
+		this.versionPredicate = versionPredicate;
+		this.restrictToMembers = restrictToMembers;
+	}
+
+	@Override
+	public Model transform(Model linkedDataModel) {
+		Model versionMaterialisedModel = ModelFactory.createDefaultModel();
+
+		Map<Resource, Resource> versionIDEntityIDMap = getVersionIDEntityIDMap(linkedDataModel, versionPredicate);
+
+		linkedDataModel.listStatements().forEach(statement -> {
```

Comment:
I find this method a bit difficult to read when you define methods at the top and then change some on the way to finally use them at the end. I feel that it is more maintainable when you make them immutable. Maybe split them in separate methods?